hack/generator/pkg/codegen/pipeline_determine_resource_ownership.go

```diff
@@ -0,0 +1,293 @@
+			}
+
+			childResourceTypeNames, err := extractChildResourceTypeNames(*childResourcePropertyTypeDef)
+			if err != nil {
+				return nil, err
+			}
+
+			err = updateChildResourceDefinitionsWithOwner(definitions, childResourceTypeNames, def.Name(), updatedDefs)
+			if err != nil {
+				return nil, err
+			}
+
+			// Remove the resources property from the owning resource spec
+			specDef = specDef.WithType(specType.WithoutProperty(resourcesPropertyName))
+
+			updatedDefs[specDef.Name()] = specDef
+		}
+	}
+
+	setResourceGroupOwnerForResourcesWithNoOwner(definitions, updatedDefs)
+	return combineNewAndExistingDefs(definitions, updatedDefs), nil
+}
+
+func resourceSpecTypeAsObject(resourceSpecDef astmodel.TypeDefinition) (*astmodel.ObjectType, error) {
+	// There's an expectation here that the spec is a typename pointing to an object. Even if the resource
+	// uses AnyOf/OneOf to model some sort of inheritance at this point that will be rendered
+	// as an object (with properties, etc)
+	specType, ok := resourceSpecDef.Type().(*astmodel.ObjectType)
+	if !ok {
+		return nil, errors.Errorf(
+			"spec (%s) type is %T, not *astmodel.ObjectType",
+			resourceSpecDef.Name(),
+			resourceSpecDef.Type())
+	}
+
+	return specType, nil
+}
+
+func extractChildResourcePropertyTypeDef(
+	definitions astmodel.Types,
+	resourceName astmodel.TypeName,
+	resourceSpecName astmodel.TypeName,
+	specType *astmodel.ObjectType) (*astmodel.TypeDefinition, error) {
+
+	// We're looking for a magical "Resources" property - if we don't find
+	// one just move on
+	resourcesProp := findResourcesProperty(specType)
+	if resourcesProp == nil {
+		return nil, nil
+	}
+
+	// The resources property should be an array
+	resourcesPropArray, ok := resourcesProp.PropertyType().(*astmodel.ArrayType)
+	if !ok {
+		return nil, errors.Errorf(
+			"Resource %s has spec %s with Resources property whose type is %T not array",
+			resourceName,
+			resourceSpecName,
+			resourcesProp.PropertyType())
+	}
+
+	// We're really interested in the type of this array
+	resourcesPropertyTypeName, ok := resourcesPropArray.Element().(astmodel.TypeName)
+	if !ok {
+		return nil, errors.Errorf(
+			"Resource %s has spec %s with Resources property whose type is array but whose inner type is not TypeName, instead it is %T",
+			resourceName,
+			resourceSpecName,
+			resourcesPropArray.Element())
+	}
+
+	resourcesDef, ok := definitions[resourcesPropertyTypeName]
+	if !ok {
+		return nil, errors.Errorf("couldn't find definition Resources property type %s", resourcesPropertyTypeName)
+	}
+
+	return &resourcesDef, nil
+}
+
+func resolveResourcesTypeNames(
+	resourcesPropertyName astmodel.TypeName,
+	resourcesPropertyType *astmodel.ObjectType) ([]astmodel.TypeName, error) {
+	var results []astmodel.TypeName
+
+	// Each property type is a subresource type
+	for _, prop := range resourcesPropertyType.Properties() {
+		// TODO: Do we need a recursive function here since this can also be a OneOf?
```

Comment:
Removed this comment -- this was really me trying to say what I said on line ~195:
```
                // TODO: These are types that cause us trouble... a lot of them use allof inheritance.
		// TODO: I think for these we will need to walk the graph of types and do a structural
		// TODO: equality check to find the name of the actual resource, but we can't do that check
		// TODO: now because these types allOf inherit from resourceBase and the actual resources
		// TODO: being referenced do not. See:
```